include/SongEditor.h

```diff
@@ -45,15 +46,44 @@ class Song;
 class TextFloat;
 class TimeLineWidget;
 
+Q_DECLARE_METATYPE(QLinearGradient)
+
 class positionLine : public QWidget
 {
+	Q_OBJECT
+	Q_PROPERTY ( bool tailGradient READ tailGradient WRITE setTailGradient )
+	Q_PROPERTY ( QColor lineColor READ lineColor WRITE setLineColor )
 public:
-	positionLine( QWidget * parent );
+	positionLine ( QWidget * parent, ComboBoxModel * zoom, Song * song );
+	
+	int width ();
+	void go ( int x, int y );	// NOTE: Check SongEditor.cpp for usage
+	void zoomUpdate ();
+	void playStateChanged ();
+	
+	// qproperty access functions
+	bool tailGradient () const;
+	void setTailGradient ( const bool & g );
+	QColor lineColor () const;
+	void setLineColor ( const QColor & c );
 
 private:
 	void paintEvent( QPaintEvent * pe ) override;
+	
+	int m_width;
+	int m_x, m_y;	// NOTE: m_x is the playback position, not the position where the tail starts
+	
+	bool m_tailGradient;
+	QColor m_lineColor;
+	
+	// to accomodate the change in size by zoom
+	ComboBoxModel * currentZoom;
+	static const QVector<double> m_zoomLevels;
```

Comment:
For static members, you need to use `s_` instead of `m_`.

Reply:
Renamed `m_zoomLevels` to `s_zoomLevels`. I'd like to point out that it is named as `m_zoomLevels` in the Song Editor class as well (since the implementation was a copy-paste).